### PR TITLE
Refactor make.py option code and add unit test selection

### PR DESCRIPTION
**Pull request by CodyDWJones**
- make.py options are now handled by argparse, allowing options like "-compiler vs2015".
- Added a "-tests_matching <regex>" option to restrict which unit tests are run to those whose name matches the regex.
- Exposed the "-num_threads" option.
- Upgraded to Catch2 v2.5.0.

Some test names were changed to prevent a "Error regular expression found in output. Regex=[No tests ran]" error while running unit tests.

### CODE REVIEW

File: make.py [hunk 1]
```diff
@@ -287,14 +220,20 @@ def do_build(cmake_exe, options):
...
 	print('Running unit tests ...')
 	ctest_cmd = '"{}" --output-on-failure'.format(ctest_exe)
+
 	if platform.system() == 'Windows' or platform.system() == 'Darwin':
 		ctest_cmd += ' -C {}'.format(config)
 
+	if args.tests_matching:
+		ctest_cmd += ' --tests-regex {}'.format(args.tests_matching)
+
+	print(ctest_cmd)
```
Comment (nfrechette) [inline make.py]: Can we remove this debug print?

File: make.py [hunk 2]
```diff
@@ -589,59 +528,13 @@ def run_clip_regression_test(cmd_queue, completed_queue, failed_queue, failure_l
...
-	print('Miscelanous:')
-	print('  -avx: On Windows, OS X, and Linux AVX support will be enabled.')
-	print('  -help: Prints this help message.')
-
 if __name__ == "__main__":
-	options = parse_argv()
-	if options['print_help']:
-		print_help()
-		sys.exit(1)
+	args = parse_argv()
```
Comment (nfrechette) [inline make.py]: How is help printed now?

Comment (CodyDWJones) [reply make.py]: It's handled for you by argparse. The "help" action causes it to print out the following, then exit:

```
D:\Home Documents\Cody\Animation compression library\acl-test>py make.py -help
usage: make.py [-build] [-clean] [-unit_test] [-regression_test]
               [-compiler {vs2015,vs2017,android,clang4,clang5,clang6,gcc5,gcc6,gcc7,gcc8,osx,ios}]
               [-config {Debug,Release}] [-cpu {x86,x64}] [-avx] [-pop]
               [-nosimd] [-num_threads NUM_THREADS]
               [-tests_matching TESTS_MATCHING] [-help]

Actions:
  If no action is specified, on Windows, OS X, and Linux the solution/make
  files are generated. Multiple actions can be used simultaneously.

  -build
  -clean
  -unit_test
  -regression_test

Target:
  -compiler {vs2015,vs2017,android,clang4,clang5,clang6,gcc5,gcc6,gcc7,gcc8,osx,ios}
                        Defaults to the host system's default compiler
  -config {Debug,Release}
  -cpu {x86,x64}        Only supported for Windows, OS X, and Linux; defaults
                        to the host system's architecture

Miscellaneous:
  -avx                  Compile using AVX instructions on Windows, OS X, and
                        Linux
  -pop                  Compile using the POPCNT instruction
  -nosimd               Compile without SIMD instructions
  -num_threads NUM_THREADS
                        No. to use while compiling and regressing
  -tests_matching TESTS_MATCHING
                        Only run tests whose names match this regex
  -help                 Display this usage information
```

File: make.py (hunk 1 above)
Comment (CodyDWJones) [reply make.py]: Of course!

**Pull request merged**